Makefile

```diff
@@ -588,7 +588,7 @@ check-all: check-kubernetes-yaml
 
 .PHONY: check-component-features
 check-component-features: ## Check that all component features are setup properly
-	${MAYBE_ENVIRONMENT_EXEC} ./scripts/check-component-features.sh
+	${MAYBE_ENVIRONMENT_EXEC} cargo hack check --each-feature --exclude-features "sources-utils-http sources-utils-tcp-keepalive sources-utils-tcp-socket sources-utils-tls sources-utils-udp sources-utils-unix sinks-utils-udp"
```

Comment:
👍 to excluding these and getting this in. Would you mind opening an issue with the final set?

Reply:
I believe these are permanent. At least, that's how I intended it. The 'util' feature pattern strikes me as reasonable, though I could be swayed to the notion that instead of having `#[cfg(feature = "some_util")]` in the code base we'd rather have `#[cfg(any(feature = "a_source", feature = "another_source", ...))]` if you feel strongly about it.

Reply:
Is the issue just dead code warnings? Maybe we could make things `pub` rather than `pub(crate)` for those modules? I think I do like `#[cfg(any(feature = "a_source", feature = "another_source", ...))]`.

Reply:
> Is the issue just dead code warnings? Maybe we could make things pub rather than pub(crate) for those modules? 

It's unused exports, yeah, but within the crate IIRC. 

> I think I do like `#[cfg(any(feature = "a_source", feature = "another_source", ...))]`.

That'd be a fair thing to take a swing at, see how we like it, in another PR. I don't think the list would be terribly long and we could drop some features from our list.